### PR TITLE
Fix severe clipping when saving generated audio

### PR DESCRIPTION
**Pull request by kklem0**
### The bug

`infer_v2.py` scales the waveform into int16 range and then hands the int16 tensor to `torchaudio.save()`:

```python
wav = torch.clamp(32767 * wav, -32767.0, 32767.0)
...
torchaudio.save(output_path, wav.type(torch.int16), sampling_rate)
```

`torchaudio.save()` treats an int16 tensor as if it were already normalised to `[-1, 1]` and scales by 32768 again, so everything past roughly 1/32768 of full scale rails. Every clip written to a file is affected — the in-memory Gradio path (`yield (sampling_rate, wav_data)`) is fine, which is why this is easy to miss.

### Measurement

A half-scale 440 Hz sine, saved both ways on torchaudio 2.11:

| | peak | samples at full scale |
|---|---|---|
| `wav.type(torch.int16)` | 1.0000 | 22043 / 22050 |
| normalised to `[-1, 1]` first | 0.5000 | 0 |

The first is a square wave.

### The fix

Normalise back to `[-1, 1]` before saving, and add the channel dimension `torchaudio` expects. Verified on an RTX PRO 6000 with CUDA 13: a real synthesis returns peak 0.511 with the patch and would rail to 1.000 without it.

We hit this running IndexTTS 2 in a dubbing pipeline, where it affected every generated segment.

### CODE REVIEW

**Comment by nanaoto** [comment]
Thanks for the detailed measurement — but I can't reproduce this on the version we pin, and I think the difference is your torchaudio version rather than the code.

`pyproject.toml:58` pins `torchaudio==2.8.*`. On 2.8.0, running your exact test (half-scale 440 Hz sine, `torch.clamp(32767 * wav, ...)` then saved both ways):

| | peak | samples at full scale | file |
| --- | --- | --- | --- |
| `wav.type(torch.int16)` (current) | 0.5000 | 0 / 22050 | PCM_16, 44144 bytes |
| normalised to `[-1, 1]` (this PR) | 0.5000 | 0 / 22050 | **FLOAT**, 88280 bytes |

Real synthesis agrees. Two files produced through `infer_v2_5.py:887` on this machine came out at peak **0.6153** and **0.4578**, with zero railed samples — not square waves.

Your report says torchaudio 2.11, and I think that's the crux. torchaudio 2.8 itself warns about exactly this area:

```
In 2.9, this function's implementation will be changed to use
torchaudio.save_with_torchcodec under the hood. Some parameters like format,
encoding, bits_per_sample ... will be ignored.
```

So the int16 handling most likely changed with that backend swap in 2.9. I'd read your finding as **real on torchaudio >= 2.9, and not triggerable under the current pin** — which also explains why nobody else hit it.

That matters for the fix, because as written it would change behaviour on 2.8 rather than preserve it: passing a float tensor makes torchaudio write a 32-bit FLOAT wav instead of PCM_16, doubling file size and changing the output format for every existing user. And `encoding` / `bits_per_sample` — the way you'd normally force PCM_16 back — are among the parameters the 2.9 backend drops, so that workaround doesn't survive the version it's meant for.

Two things that would help move this forward:

1. Can you confirm whether it reproduces on `torchaudio==2.8.*`? If it doesn't, the fix should be conditional on the version (or on the backend), not unconditional.
2. `indextts/infer_v2_5.py` has the same `torchaudio.save(output_path, wav.type(torch.int16), sampling_rate)` at `:887`, plus another save at `:541`. 2.5 is the current default path, so a v2-only fix would leave most users on the affected code once the pin moves past 2.9.

Not closing this — if the pin ever moves to 2.9+, this becomes a live bug and your analysis is what will fix it. I'd just rather not change the 2.8 output format to pre-empt it.
